src/values/Value.js

```diff
@@ -34,6 +34,7 @@ export default class Value {
     switch (typeName) {
       case "empty": return EmptyValue;
       case "undefined": return UndefinedValue;
```

Comment:
The first issue was that when using the string "void" for residual functions it would return undefined instead of UndefinedValue for the type. What is the reason for removing "undefined" case? It may be used by other portions of the code which would then crash?